include/runtime_symbol_lookup.hpp

```diff
@@ -9,26 +9,85 @@
 #include "symbol_map.hpp"
 #include "symbol_table.hpp"
 
+#include "ddres_def.hpp"
+#include <array>
 #include <unordered_map>
 
 namespace ddprof {
 
 class RuntimeSymbolLookup {
 public:
   explicit RuntimeSymbolLookup(std::string_view path_to_proc)
-      : _path_to_proc(path_to_proc) {}
+      : _path_to_proc(path_to_proc), _cycle_counter(1) {}
+
+  SymbolIdx_t get_or_insert_jitdump(pid_t pid, ProcessAddress_t pc,
+                                    SymbolTable &symbol_table,
+                                    std::string_view jitdump_path);
+
   SymbolIdx_t get_or_insert(pid_t pid, ProcessAddress_t pc,
                             SymbolTable &symbol_table);
+
   void erase(pid_t pid) { _pid_map.erase(pid); }
 
+  void cycle() { ++_cycle_counter; }
+
 private:
-  using PidUnorderedMap = std::unordered_map<pid_t, SymbolMap>;
-  void fill_perfmap_from_file(int pid, SymbolMap &symbol_map,
-                              SymbolTable &symbol_table);
+  using FailedCycle = std::unordered_map<std::string, uint32_t>;
```

Comment:
You could use transparent comparator here to avoid creating a `std::string` at each call to the map (you will be able to use `std::string_view` directly), with some tricks (it's a bit painful for `std::unordered_map` although easier for std::map).

Have a look at [this](https://www.reddit.com/r/cpp/comments/phqxiq/comment/hbkis8h/?utm_source=share&utm_medium=web2x&context=3) and [this godbolt link](https://godbolt.org/z/hqYxx868W) to make it compile.

Reply:
I really liked this comment ! This was bugging me. C++ still makes it a little complex though ^^